plaid/sandbox/client.py

```diff
@@ -21,138 +22,128 @@ def json(self):
         raise NotImplementedError
 
-    @staticmethod
-    def post_historical_webhook(url, data):
+    @abc.abstractmethod
+    def post_historical_webhook(self, url, data):
         raise NotImplementedError
 
-    def get_account(self, access_token=None):
-        access_token = access_token or self.access_token
-        return json.loads(access_token[5:])
-
-    def gen_access_token(self, account):
-        self.access_token = 'test_%s' % json.dumps(account, sort_keys=True)
-        return self.access_token
+    def _post_webhooks(self):
+        url = self._account.get('webhook')
+        if url is not None:
+            init_data = self._load_fixture('webhook/initial.json')
+            hist_data = self._load_fixture('webhook/historical.json')
+            self.post_initial_webhook(url, init_data)
+            self.post_historical_webhook(url, hist_data)
 
-    @staticmethod
-    def _load_fixture(filename):
+    def _load_fixture(self, filename):
         dir_name = os.path.dirname(__file__)
         with open(os.path.join(dir_name, 'fixtures', filename)) as f:
-            return json.loads(f.read())
-
-    @classmethod
-    def _post_webhook(cls, account, event_type):
-        data = SandboxClient._load_fixture('webhook/%s.json' % event_type)
-        data['access_token'] = 'test_%s' % json.dumps(account, sort_keys=True)
-        if event_type == 'initial':
-            cls.post_initial_webhook(account['webhook'], data)
-        elif event_type == 'historical':
-            cls.post_historical_webhook(account['webhook'], data)
-
-    def _process_connect_success(self, data, account=None):
-        account = account or self.get_account()
+            data = json.loads(f.read())
 
-        if account.get('login_only'):
-            del data['transactions']
-        if account.get('webhook') is not None:
-            self._post_webhook(account, 'initial')
-            self._post_webhook(account, 'historical')
+        if 'access_token' in data:
+            data['access_token'] = self.get_access_token()
+        if 'accounts' in data:
+            account_type = self._account['account_type']
+            for a in data['accounts']:
+                a['_id'] = account_type + a['_id']
 
-    def _load_connect_success(self, account_type=None):
-        if not account_type:
-            account_type = self.get_account()['account_type']
+        return data
 
+    def _load_connect_success(self, keep_transactions=False):
+        account_type = self._account['account_type']
         filename = {
             'wells': 'connect/no_transactions.json',
         }.get(account_type, 'connect/success.json')
-        return self._load_fixture(filename)
+
+        data = self._load_fixture(filename)
+        if not keep_transactions and self._account.get('login_only'):
+            del data['transactions']
+        return data
+
+    def set_access_token(self, access_token):
+        self._account = json.loads(self.access_token[5:])
```

Comment:
Should this be `access_token[5:]` instead of `self.access_token[5:]`? Also where is `set_access_token ` being called from?